### DISCUSSION TITLE
Trouble open custom view controller view in plugin.

### DISCUSSION THREAD

**Discussion by jaybowman** [ionic-team/capacitor#6054] [Q&A]
I'm writing a custom navigation plugin using ARCGis iOS sdk. Our previous version of the app used Cordova, so the swift code is tested and works. I followed the instructions in the capacitor docs on how to open a view controller from your plugin code. What's happening is that the simulator screen goes black when I call the viewController from the plugin. Execution returns from the plugin method. I have breakpoints set in the controller viewDidLoad() and viewDidAppear(), but the view never displays. My @IBOutlet variables are nil. So, it seams like there is a disconnect between my controller and the xib file. But everything looks good in the Xcode Interface builder.
 I used an UIAlertController in place of my NavigateRouteViewController and the alert box appears on top of my ionic web view as expected.

```
@objc(MapNavPlugin)
public class MapNavPlugin: CAPPlugin {
...

    @objc func startNavigation(_ call: CAPPluginCall) {
     ...
        DispatchQueue.main.async {
             let mapCtrl = NavigateRouteViewController(withRouteInfo: routeInfo)
              mapCtrl.modalPresentationStyle = .fullScreen
              self.bridge!.viewController!.present(mapCtrl, animated: true, completion: nil)
              
               let info: String = "Hello from main task for debugging"
               print(info)
          }
    }
...
}
```

**Comment by jcesarmobile** [comment]
There isn't enough information, but my guess is, since Capacitor plugins are CocoaPods libraries, you are not distributing the .xib file correctly.
You should take a look on the [resources](https://guides.cocoapods.org/syntax/podspec.html#resources) and [resource_bundles](https://guides.cocoapods.org/syntax/podspec.html#resource_bundles) properties of the Podspec Syntax Reference docs.

**Reply by jaybowman** [reply]
Thanks for the nudge in the right direction.

**Comment by jaybowman** [comment]
With much trial and error. Here is my solution: First I needed to add the resource_bundle to the pod spec
 s.resource_bundles = {'CapacitorPluginArcmap' => ['ios/Plugin/*.xib'] }
Then in my view controller init() 
```
        let pluginBundle = Bundle( for: NavigateRouteViewController.self )
        let pluginResourceURL = pluginBundle.resourceURL?.appendingPathComponent("CapacitorPluginArcmap.bundle")
        let resourceBundle = Bundle(url: pluginResourceURL!)
        super.init(nibName: "NavigateRouteViewController", bundle: resourceBundle)
```
